File: Prefix Sum/Problem1 - 1732.java
public class Main {
    public static int largestAltitude(int[] gain) {
        int n = gain.length;
        int maxGain = 0;
        int currGain = 0;
        for (int i = 0; i < n; i++) {
            currGain += gain[i];
            maxGain = Math.max(maxGain, currGain);
        }
        return maxGain;
    }

    public static void main(String[] args) {
        int[] gain = {-5, 1, 5, 0, -7};
        System.out.println(largestAltitude(gain));
    }
}

File: Prefix Sum/Problem1 - 1732.cpp
#include <bits/stdc++.h>
using namespace std;

/* brute force approach - using extra prefix array */

/* optimal approach - using prefix sum */
int largestAltitude(vector<int>& gain) {
    int n = gain.size();
    int maxGain = 0;
    int currGain = 0;
    for (int i = 0; i < n; i++) {
        currGain += gain[i];
        maxGain = max(maxGain, currGain);
    }
    return maxGain;
    /*  TC-O(n) SC-O(1)  */
}